Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs

```diff
@@ -2619,6 +2624,41 @@ private SpecialProcessKind GetProcessKind(AbsolutePath processOverride)
             return s_specialTools.TryGetValue(toolName.ToLowerInvariant(), out SpecialProcessKind kind) ? kind : SpecialProcessKind.NotSpecial;
         }
 
+        private static bool StringLooksLikeBuildXLTestTempFile(string fileName)
+        {
+            if (!fileName.EndsWith(".tmp", StringComparison.OrdinalIgnoreCase))
+            {
+                return false;
+            }
+
+            int beginCharIndex = fileName.LastIndexOf('\\');
+
+            if (beginCharIndex == -1 || beginCharIndex + 3 >= fileName.Length)
+            {
+                return false;
+            }
+
+            char c1 = fileName[beginCharIndex + 1];
+            if (c1.ToUpperInvariantFast() != 'B')
+            {
+                return false;
+            }
+
+            char c2 = fileName[beginCharIndex + 2];
+            if (c2.ToUpperInvariantFast() != 'X')
+            {
+                return false;
+            }
+
+            char c3 = fileName[beginCharIndex + 3];
+            if (c3.ToUpperInvariantFast() != 'L')
+            {
+                return false;
+            }
+
+            return true;
+        }
+
         private static bool StringLooksLikeRCTempFile(string fileName)

```

Comment:
>StringLooksLikeRCTempFile [](start = 28, length = 25)

I'm really not a fan of these special cased exceptions in SandboxedProcessPipExecutor. You're going the right direction in making this one less likely to be exercised but I'd love to move more in the direction of not utilizing them at all. For shared opaques, it seems like we shouldn't really even need them

Reply:
I know you're not a big fan of this exception. We can add a field in the configuration file describing the tool and the paths to ignore. But I guess that's beyond the purpose of this PR. We can add a user story to have a proper implementation.

Yes, shared opaque directories do not need this change in Detours. This change is for completeness.

---
In reply to: [314459791](https://github.com/microsoft/BuildXL/pull/741#discussion_r314459791) [](ancestors = 314459791)